src/babel.js

```diff
@@ -294,19 +307,29 @@ export default function({ types: t }) {
           // Transpile external styles
           path.traverse(externalStylesVisitor, state)
         },
-        exit({ scope }, state) {
+        exit(path, state) {
+          // For source that didn't really use styled-jsx imports,
+          // remove the injected import at the beginning
+          if (!state.file.hasJSXStyle && !state.file.hasCssImports) {
+            path.traverse({
+              ImportDeclaration(importPath) {
+                if (importPath.node.source.value === state.styleModule) {
+                  importPath.remove()
+                }
+              }
```

Comment:
In CommonJS modules `addDefault` will not add ES6 imports and this visitor is not going to remove them. 

Do you know if `addDefault` returns a node or a path? If it returns the import statement or import specifier object (a `path`) that it injected you can store that in `state` and in Program.exit remove that path or closest import path.

Reply:
`addDefault` returns a node of identifier type so no luck to delete conveniently.

Regarding to commonjs I'm not sure which case are you referring here. Do you mean if babel `option.sourceType` is `'unambiguous'`?

Reply:
I'm not sure if pure commonjs source should be considered. Like we only handle the import declarations in babel-external. 

```js
const { css } = require("styled-jsx/css")
exports.name = css`div { color: red; }
```

If user writes the code above, plugin won't handle it either.

Reply:
Oh ok! I'd need to look at the implementation because I don't remember if we support CommonJS. But if you can confirm that we don't then we don't even need to use addDefault and we can add the import ourselves like we were doing it before avoiding to insert the import and the remove it.

Reply:
Yeah, this inconsistence was introduced in #684 but `styled-jsx/css` imports are not effected.

I wonder if we could ship this first since it will only inject unused require calls in pure commonjs sources with jsx style usage, and it could be tree-shaked by bundler.

This can resolve the current issue of "conflicted _JSXStyle name" and "undefined _JSXStyle" issues. (Also unblock some of react 18 integration work 😂.) 

I can file a PR later to change it back to previous imports injection if this plan sounds good to you.

Reply:
Sounds good!